CrimeMonitor/CrimeMonitor.cs

```diff
@@ -0,0 +1,849 @@
+        }
+
+        public void Stop()
+        {
+            running = false;
+        }
+
+        public void Reload()
+        {
+            readRecord();
+            Logging.Info("Reloaded " + MonitorName() + " " + MonitorVersion());
+
+        }
+
+        public void _start()
+        {
+            running = true;
+
+            while (running)
+            {
+                List<FactionRecord> recordList;
+                lock (recordLock)
+                {
+                    recordList = criminalrecord.ToList();
+                }
+
+                Thread.Sleep(5000);
+            }
+        }
+
+        public UserControl ConfigurationTabItem()
+        {
+            return new ConfigurationWindow();
+        }
+
+        public void EnableConfigBinding(MainWindow configWindow)
+        {
+            configWindow.Dispatcher.Invoke(() => { BindingOperations.EnableCollectionSynchronization(criminalrecord, recordLock); });
+        }
+
+        public void DisableConfigBinding(MainWindow configWindow)
+        {
+            configWindow.Dispatcher.Invoke(() => { BindingOperations.DisableCollectionSynchronization(criminalrecord); });
+        }
+
+        public void HandleProfile(JObject profile)
+        {
+        }
+
+        public void PostHandle(Event @event)
+        {
+        }
+
+        public void PreHandle(Event @event)
+        {
+            Logging.Debug("Received event " + JsonConvert.SerializeObject(@event));
+
+            // Handle the events that we care about
+            if (@event is BondAwardedEvent)
+            {
+                handleBondAwardedEvent((BondAwardedEvent)@event);
+            }
+            else if (@event is BondRedeemedEvent)
+            {
+                handleBondRedeemedEvent((BondRedeemedEvent)@event);
+            }
+            else if (@event is BountyAwardedEvent)
+            {
+                handleBountyAwardedEvent((BountyAwardedEvent)@event);
+            }
+            else if (@event is BountyIncurredEvent)
+            {
+                handleBountyIncurredEvent((BountyIncurredEvent)@event);
+            }
+            else if (@event is BountyPaidEvent)
+            {
+                handleBountyPaidEvent((BountyPaidEvent)@event);
+            }
+            else if (@event is BountyRedeemedEvent)
+            {
+                handleBountyRedeemedEvent((BountyRedeemedEvent)@event);
+            }
+            else if (@event is FineIncurredEvent)
+            {
+                handleFineIncurredEvent((FineIncurredEvent)@event);
+            }
+            else if (@event is FinePaidEvent)
+            {
+                handleFinePaidEvent((FinePaidEvent)@event);
+            }
+            else if (@event is DiedEvent)
+            {
+                handleDiedEvent((DiedEvent)@event);
+            }
+
+        }
+
+        private void handleBondAwardedEvent(BondAwardedEvent @event)
+        {
+            if (@event.timestamp > updateDat)
```

Comment:
Move this if statement to the PreHandle method (rather than repeating it for each individual event)?

Reply:
Concur. Done.